src/AbstractArray.php

```diff
@@ -357,6 +357,22 @@ public function isEmpty()
     }
 
     /**
+     * Check whether array is associative or not
+     *
+     * @return bool Returns true if associative, false otherwise
+     */
+    public function isAssoc()
+    {
+        foreach ($this->getKeys() as $key) {
+            if (!is_int($key)) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
@m1 My only point is that I don't consider next array as associative:
```php
[
    1 => 'one',
    'two' => 2,
    3 => 'three',
]
```
It's a mixed (numeric and assoc).
So if we want to check whether array is only **assoc**, lets inverse this method logic to:
```php
foreach ($this->getKeys() as $key) {
    if (!is_string($key)) {
        return false;
    }
}

return true;
```
What do you think?

BTW, we could add `isNumeric()` method to check whether array is only `numeric` with:
```php
foreach ($this->getKeys() as $key) {
    if (!is_int($key)) {
        return false;
    }
}

return true;
```